riscv-c-api.md

```diff
@@ -114,7 +127,88 @@ For example:
 * `__attribute__((interrupt("supervisor")))`
 * `__attribute__((interrupt("machine")))`
 
-## Intrinsic Functions
 
-Do we really have none of these?  I can't figure out
-`gcc/gcc/config/riscv/riscv-builtins.c`...
+## Scalar Cryptography Extension Intrinsics
+
+The short intrinsics are instantiated by including the `rvkintrin.h` system
+header file.
+
+For post-ratification Scalar Cryptography v1.0, the Arch Version is `1000000`
+(one million).
+
+The extensions `Zkt` or `Zkr` do not introduce intrinsics. The data-independent
+latency is a property of the implementation, while `Zkr` (the entropy source
+interface) is implemented as a CSR only (`seed` at CSR address `0x015`).
+
+Many of the `Zbkb` instructions can be inferred compilers so they do not
+have direct intrinsics equivalents. Rotations are included mainly as RISC-V
+sign extension behavior, and clamping of rotation amount may generate extra
+instructions unless the programmer is very careful.
+
+### Scalar Cryptography Extension short form intrinsics (alphabetically)
+
+| Prototype                                                     | Mnemonic      | Short Description                         | Supported in                  |
+| ------------------------------------------------------------- | ------------- | ----------------------------------------- | ----------------------------- |
+| `int32_t _rv32_aes32dsi(int32_t rs1, int32_t rs2, int bs);`   | `aes32dsi`    | AES final round decryption / RV32.        | Zknd, Zkn, Zk (RV32)          |
```

Comment:
Added the basic rules back in:

"The prefixes and data types are:

* `_rv_*(...)`: intrinsics that operate on the `long` data type.
* `_rv32_*(...)`: intrinsics that operate on the `int32_t` data type.
* `_rv64_*(...)`: RV64-only intrinsics that operate on the `int64_t` data type.
 "

Reply:
See also: https://github.com/riscv-non-isa/riscv-c-api-doc/pull/25

Reply:
Perhaps I missed something, but there's nothing useful at https://github.com/riscv-non-isa/riscv-c-api-doc/pull/25 . The convention here is the one originally Claire defined for Bitmanip's "rvintrin.h", and which has been floating around for several years.

I personally don't want to define a new convention. The way I see it, If you want to define a new convention, then define a new convention, document it well, and explain why you did that. Don't expect other people to halt everything to try to guess what is it that you want.

Reply:
Well, I guess you understand the common naming conventions have something useful.

It might have offended you that I did not reuse the ones that you are using. That was not my intention (apologies, if that is the case).

My proposal was based on the naming convention from the [vector intrinsics](https://github.com/riscv-non-isa/rvv-intrinsic-doc/blob/master/rvv-intrinsic-rfc.md#naming-rules), because they are already implemented in upstream LLVM.

Reply:
Well, these are the common conventions we have used with the Scalar instructions. I did not invent them; I believe Claire did.